File: python_assignments_and_projects/Users with Banking Account/BankAccount.py
class BankAccount:
    def __init__(self,name,balance,int_rate):
        self.name = name
        self.balance = balance
        self.interest = int_rate
# deposit(self, amount) - increases the account balance by the given amount

    def deposit(self,amount):
        self.balance += amount
        return self
    
    # withdraw(self, amount) - decreases the account balance by the given amount if there are sufficient

    def withdraw(self,amount):
        if self.balance >= amount:
            self.balance -= amount
            return self
        else:
            print('Insuficient Funds: Charging a $5 fee')
            self.balance -= 5
            return self
# display_account_info(self) - print to the console: eg. "Balance: $100"

    def display_account_info(self):
        print(f"Name:{self.name}  BALANCE :{self.balance}")
        return self
    def yield_interest(self):
        self.balance += self.balance * self.interest
        return self

chaker= BankAccount("Chaker",500,0.05)
jalil= BankAccount("Jalil",10000, 0.20)
aws = BankAccount("AWS",2,0.2)

chaker.deposit(5000).deposit(1000).deposit(50000).withdraw(15000).yield_interest().display_account_info()
jalil.deposit(7000).deposit(9000).withdraw(100).withdraw(100).withdraw(100).withdraw(100).yield_interest().display_account_info()
aws.withdraw(5).display_account_info()

    

File: Java/Java Assignments/BankAccount/src/com/axsosacademy/bankaccount/BankAccount.java
package com.axsosacademy.bankaccount;

public class BankAccount {
    // MEMBER VARIABLES
    private double checkingBalance;
    private double savingsBalance;

    private static int accounts;
    private static double totalMoney; // refers to the sum of all bank account checking and savings balances

    // CONSTRUCTOR
    // Be sure to increment the number of accounts
    public BankAccount() {
    	accounts ++;
    }
    
    // GETTERS
    // for checking, savings, accounts, and totalMoney
    public double getCheckingBalance() {
    	return checkingBalance;
    }
    public double getSavingsBalance() {
    	return savingsBalance;
    }
    public double getAccounts() {
    	return accounts;
    }
    public double getTotalMoney() {
    	return totalMoney;
    }
    // METHODS
    // deposit
    // - users should be able to deposit money into their checking or savings account
    public void deposit(double amount ,  String chooseAccount)  {
    	if(chooseAccount.equals("saving")) {
    		savingsBalance+=amount;
    	}
    	else if(chooseAccount.equals("checking")) {
    		checkingBalance+=amount;
    	}
    	totalMoney+=amount;
    }
    // withdraw 
    // - users should be able to withdraw money from their checking or savings account
    // - do not allow them to withdraw money if there are insufficient funds
    // - all deposits and withdrawals should affect totalMoney
    
    public void withdraw(double amount, String chooseAccount) {
    	if(chooseAccount.equals("saving")) {
    		if (amount>savingsBalance) {
    			System.out.println("Insuficient Funds Amigo");
    			return;
    		}
    		else {
    			savingsBalance -= amount;
    		}
    	}
    		else if(chooseAccount.equals("checking")) {
    			if(amount>checkingBalance) {
    				System.out.println("Insuficient Funds Amigo");
    				return;
    			}else {
    				checkingBalance -= amount;
    		}
    	}
    }
    
    // getBalance
    // - display total balance for checking and savings of a particular bank account
    	public void getBalance() {
    		String displayChecking = String.format("This is Your Account Balance for Checking Account   %.2f  %n" ,checkingBalance);
    		System.out.println(displayChecking);
    		System.out.printf("This is Your Account Balance for saving Account  %.2f%n" ,savingsBalance);
    		System.out.printf("This is Your total Accounts Balance  %.2f%n" ,savingsBalance+checkingBalance);
    		
    
    
    
    }
}
